### PR TITLE
[WIP] New UI

### PR DESCRIPTION
**Pull request by ping**
This is the PR for the UI work I mentioned in https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/issues/217#issuecomment-381340148

Change summary:

- Incorporate the [Bulma CSS framework](https://github.com/jgthms/bulma)
- Changed many of the functions in ``action.cgi`` ``scripts.cgi`` ``status.cgi`` ``getMotionInfo.cgi`` to return json so that we can have more elegant ajax-y UI flows
- Changed the "live" view to only refresh statuses on demand (when Camera Controls menu is hovered/clicked) instead of a constant interval refresh. I did not like the constant refresh in the current implementation because it was sometimes locking up the UI in the background.
- Changed the "live" view image to reload only after previous image is loaded instead of a constant refresh interval (uses image "onload" to trigger)
- I've fixed up some bugs in configMotion.html and also made the UI work responsively
- Added a "shutdown" command (calls ``/sbin/halt``) so that I can gracefully shutdown the camera instead of pulling the plug
- Add option to choose a dark UI theme

This is tentatively still a WIP because I feel that more testing is required but I've submitted this PR as a basis for discussion/feedback since I've had to change many of the core .cgi scripts.

Edit: Latest screenshots
gh_dafang_01
gh_dafang_02
gh_dafang_03
gh_dafang_04
gh_dafang_05
gh_dafang_06

### CODE REVIEW

File: firmware_mod/www/cgi-bin/action.cgi [hunk 1]
```diff
@@ -121,45 +127,45 @@ if [ -n "$F_cmd" ]; then
...
       if [ "$(cat /etc/TZ)" != "$tz" ]; then
-        echo "Setting TZ to '$tz'...<br/>"
+        echo "<p>Setting TZ to '$tz'...</p>"
         echo "$tz" > /etc/TZ
-        echo "Syncing time...<br/>"
-        if [ "$(/system/sdcard/bin/busybox ntpd -q -n -p $conf_ntp_srv 2>&1)" -eq 0 ]; then
-          echo "<br/>Success<br/>"
-        else echo "<br/>Failed<br/>"
+        echo "<p>Syncing time...</p>"
+        if [ "$(/system/sdcard/bin/busybox ntpd -q -n -p time.google.com 2>&1)" -eq 0 ]; then
```
Comment (jmtatsch) [inline firmware_mod/www/cgi-bin/action.cgi]: this may break ntp configuration via file?

Comment (ping) [reply firmware_mod/www/cgi-bin/action.cgi]: @jmtatsch Fixed. Thanks for the catch. Turns out that I missed out on the ntp form in the ``status.cgi`` as well.

**Comment by gysmo38** [comment]
I test your UI, it is so great :) better than mine :p 
Some comments:
- I find better to have ptz button (left, right, top , down) directly on picture
- in camera controls the button not reflect the status of the camera
- on my laptop the picture is little to be so I need to scroll down to have view it
- maybe add camera name in title could be great

If I can help let me knwon

**Comment by jmtatsch** [comment]
I would also love gysmo’s on picture motion controls in the mix

**Comment by gysmo38** [comment]
@ping 
- great :)  maybe if it possible to put the pad at right bottom corner. Like in tiny cam :p 
- I made more test with button in camera controls. All button are off for me. If I toggle one, the command work but after 1 or 2 seconds it come back to off. I check, I have no javascript error.
- Thank the live view is better. I have scrolldown for the bottom page. Is it possible to reduce space at the top to have full page on the screen?
- great :)

**Comment by ping** [comment]
@gysmo38 
- Never used tinycam so I've no idea about their UI. It's difficult to pin the dpad flushed to the side of the image because the image is responsive. If you know how to, please let me know.
- Can you check a few things? Open up the Developer Console => Network tab, then do a hard refresh (control+F5 I think, or control+click on refresh). Do you see the requests being made below?
Does opening http://<camip:port>/cgi-bin/state.cgi?cmd=yellow_led in a new tab give the correct "on"/"off" response?
I've patched the cgi scripts to always return no-cache in case browser caching was causing problems, so please refresh your version again.

dafang_debug_network

**Comment by MkLHX** [comment]
Hi there!
I'm propose an other ui for only html files in the project by using https://getmdl.io/ framework

So i submit a PR #254 

If i can help for a new responsive ui i'm in!

**Comment by EliasKotlyar** [comment]
Hello everyone,

Many thanks contributions for the new UI. I have tested it, and its looking great. :) 

Greatings
Elias

**Comment by gysmo38** [comment]
@ping I tested with the commit. It work for me :)
I will try to move pad. 
Another idea, for rebooting have a coutdown (maybe 30s) before refresh page.

**Pull request merged**